### PR TITLE
Domains Checkout: Improve contact form accessibility

### LINKED ISSUE #21660: Checkout: Accessibility issues with domain contact checkout form
Testing with VoiceOver + Safari, and keyboard-only + Chrome. I had some trouble filling out the domain contact information form:

screen shot 2018-01-18 at 1 23 53 pm

> Phone

This read my phone number as "five hundred fifty five" as I entered it, rather than "five five five". Not really a bug, but unexpected. Might be fixed by changing the input type?

> Country

If I tab from Phone to Country, and start typing, this field auto-selects the first country with that letter, and pushes me into the next field. If I start by using the arrow-key to open the dropdown, I can then type the full country name and it selects the correct country. This is what I expect to happen by default. The state field does this as I expect.

> Postal Code

In my testing of VoiceOver, I ended up selecting the wrong country, so the form wouldn't let me continue with the postal code I entered. This error wasn't announced to me, though, so I couldn't figure out why I couldn't finish checkout. See this example of [using aria roles](https://www.w3.org/TR/wai-aria-practices/examples/alert/index.html) for one way to solve this.

### PR DESCRIPTION
**Pull request by Copons**
Fix #21660

- Currently, upon selecting a country from the country list, the focus is automatically moved to the following field.
Even though I'm opposed to this behaviour, it could still make sense when the form is controlled via mouse or touch.
But, it makes impossible to select a country by typing its name.
E.g. If I wanted to select `Italy`, I would type `i` and then `t`. Though, `i` immediately selects the first country name starting with that letter, `Iceland`, and automatically moves the focus to the address field.
This PR removes this behaviour, making the country select field as standard as possible.

- The postal code field is validated (on change) against the country.
E.g. Italian postal codes only allow numbers, so typing `00100a` would display a validation error.
This error is not read by the screen reader, so visually impaired users do not receive any feedbacks that there are errors in the form.
This PR adds a `role="alert"` to the error, which is immediately read by the screen reader.
This change is not limited to the Domains Checkout form, but to **all** `FormInputValidation` usages.

## Note

#21660 also reported a bug with the phone field:
> This read my phone number as "five hundred fifty five" as I entered it, rather than "five five five". Not really a bug, but unexpected. Might be fixed by changing the input type?

I cannot reproduce this (VoiceOver reads my own number starting with `333` as `three three three`.
@ryelle can you confirm it? I'm using Mac's VoiceOver on macOS 10.13.3, and Chrome 64.

Though, I've found different issues related to the country code:

- The phone country code select field is not focusable via keyboard ([notice the `tabindex="-1"`](https://github.com/Automattic/wp-calypso/blob/master/client/components/phone-input/index.jsx#L254)).
- The phone country code select field overrides whatever I write in the phone text field.
In other words, if the phone country is set to "USA", the text field forces a `+1` into the text field.
This becomes very problematic keyboard users, because they can't actually change the phone country.

This is a more complicated change, which requires its own PR, so it won't be tackled here.

## Testing instructions

- Purchase a new domain for a site.
- At the checkout screen make sure that:
  - Selecting a country does not automatically move the focus to the address field.
  - Enabling VoiceOver, an error message is read by the screen reader upon entering an incorrect postal code.

### CODE REVIEW

**Comment by ryelle** [comment]
> can you confirm it?

No, it didn't happen this time - I think it has to do with how VoiceOver understands what is a number, and possibly with the verbosity settings of VoiceOver. So, probably not a "real" issue after all.
